client/v3/balancer/balancer.go

```diff
@@ -191,6 +216,24 @@ func (bb *baseBalancer) HandleResolvedAddrs(addrs []resolver.Address, err error)
 			// (DO NOT) delete(bb.scToSt, sc)
 		}
 	}
+
+	// TODO: Consider just returning `ErrBadResolverState` if `warnedErrors` is
+	//       not empty. Or (to make a breaking change) just early exit with a
+	//       `ErrBadResolverState` when `warnedErrors` would first be appended to.
+	return coalesceErrors(warnedErrors)
+}
+
+func coalesceErrors(errors []error) error {
+	if len(errors) == 0 {
+		return nil
+	}
+
+	return errors[0]
```

Comment:
@ptabor For now I did the most basic thing (grab the first error) to coalesce, because I didn't want to (A) copy-paste in some quasi-bloated multi-error struct to support here or (B) take on a dependency without asking.

However, note that https://pkg.go.dev/go.uber.org/multierr is already a dependency in the `go.sum` for `client/v3` (and most other subpackages in this repo).

https://pkg.go.dev/github.com/hashicorp/go-multierror is also an option if we want to go down the dependency path

Reply:
Sounds good. It was been updated.